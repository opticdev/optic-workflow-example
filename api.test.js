const fetch = require('node-fetch');

test('Retrieve existing user', async () => {
  const result = await fetch(api('/users/jdoe')).then((res) => res.json());
  expect(result).toMatchSnapshot();
});

test('Retrieve unknown user', async () => {
  const result = await fetch(api('/users/unknown')).then((res) => res.json());
  expect(result).toMatchSnapshot();
});

test('Retrieve existing user', async () => {
  const result = await fetch(api('/users2/jdoe')).then((res) => res.json());
  expect(result).toMatchSnapshot();
});

function api(path) {
  return `${process.env.OPTIC_PROXY}${path}`;
}
