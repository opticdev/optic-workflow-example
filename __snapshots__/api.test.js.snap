// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Retrieve existing user 1`] = `
Object {
  "email": "jdoe@example.com",
  "name": "Jane Doe",
}
`;

exports[`Retrieve unknown user 1`] = `
Object {
  "title": "User unknown not found",
}
`;
